### PR TITLE
docs(README): Update readme for tests

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
Update the documentation for the tests and add references so the test can be runnable.

### Testing
Please check that the links are correct and everything makes sense.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
@smlng ping

File: tests/periph_i2c/README.md [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+for the PHiLIP version used.
+
+This test will require that both `sda` and `scl` pins of the `i2c.dev 0` to
+be wired to the `DUT_SDA` and `DUT_SCL` pins on PHiLIP. If the device has a
+reset pin then the optional `DUT_RST` pin on PHiLIP can be connected.
+
+DUT pin name | PHiLIP pin name | Required
+-------------|-----------------|---------
+sda          | DUT_SDA         | Yes
+scl          | DUT_SCL         | Yes
```
Comment (smlng) [inline tests/periph_i2c/README.md]: shouldn't they also share at least one `GND`?

File: tests/periph_i2c/README.md [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
-using the RIOT shell.  Consult the 'help' shell command for available actions.
+using the RIOT shell. The `robot-test` feature requires a target board to be
+connected to the hardware reference board PHiLIP.
 
-## Background
+## Automated Test Setup
 
-Test for the low-level I2C driver.  Can be run manually or using automated test
-script for RobotFramework, run `make robot-test`.
+Before executing the automated test, the DUT (device under test) and reference
```
Comment (smlng) [inline tests/periph_i2c/README.md]: DUT is in the test glossary, no need to write it out here

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+## Glossary
+
+DUT - Device Under Test
+PHiLIP - The reference hardware to test against
+
+## make overrides
+
+Variable     | Description                                     | Example(s)
+-------------|-------------------------------------------------|------------------------------------
+BOARD        | The name of the DUT board _(required)_          | nucleo-f030r8, samr21-xpro
```
Comment (smlng) [inline tests/README.md]: I would add 2 columns: 1.) required (yes/no), 2.) default value

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+PHILIP_PORT  | The serial port of PHiLIP _(required)_          | /dev/ttyACM0
+CMD_TIMEOUT  | The time (s) for a command to respond           | 0.5, 1
+RESET_WAIT   | The time (s) to wait after a reset              | 0, 3
+CONNECT_WAIT | The time (s) to wait after connecting to serial | 0, 5
+RESET        | The command used to reset the device            | 'python3 -m philip_pal --dut_reset'
+RESET_FLAGS  | Flags for the reset command                     | /dev/ttyACM0, ${PHILIP_PORT}
+
+The `RESET_WAIT` is useful for speeding up tests if a longer reset wait period
+is not needed like in the case of using the PHiLIP reset. Then `RESET_WAIT`
+can be 0. For some boards, such as the arduino, the bootloader is activated
```
Comment (smlng) [inline tests/README.md]: `the Arduino(s)`, start with capital as it is a name or trademark ?

File: tests/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Robot Framework Automated Hardware Tests
+
+These are tests that are run with Robot Framework and involve a shell
+interactions and usually using reference hardware.
+
+## Glossary
+
+DUT - Device Under Test
+PHiLIP - The reference hardware to test against
```
Comment (smlng) [inline tests/README.md]: add a link to Philip repo?

File: tests/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Robot Framework Automated Hardware Tests
+
+These are tests that are run with Robot Framework and involve a shell
```
Comment (smlng) [inline tests/README.md]: involve ~~a~~ shell interactions

File: README.md [hunk 7]
```diff
@@ -41,6 +41,10 @@ Besides the applications that are written in C, it is recommend to use Python
 to write tools or interfaces related to running tests. That allows for simple
 integration with the RF test scripts (and re-usability in general).
 
+## Testing a Pull Request
```
Comment (smlng) [inline README.md]: I would rephrase (or generalise) here, bc its not only about testing a PR (which you btw. cannot simply checkout within the submodule without changing its git config, too). Also what PR? From RIOT-OS or RobotFW-tests 😮

So its more about testing a different version of RIOT, i.e. not the current submodule state. So you can either checkout that RIOT version somewhere else and overwrite `RIOTBASE` or update the RIOT submodule to the desired commit, however for the latter you cannot simply switch to a PR.

File: tests/periph_i2c/README.md [hunk 8]
```diff
@@ -1,17 +1,50 @@
...
+
+## Running Automated Tests
+
+After the setup is complete use the following command to execute the robot test:
+
+`PHILIP_PORT=<philip_serial_port> BOARD=<DUT_BOARD_NAME> make flash robot-test`
+
+### Configuring run parameters
+
+A number of setting can be adjusted such as changing timeouts or using the
```
Comment (smlng) [inline tests/periph_i2c/README.md]: setting[s]?

File: tests/README.md (hunk 3 above)
Comment (MrKevinWeiss) [reply tests/README.md]: OK

File: tests/README.md (hunk 4 above)
Comment (MrKevinWeiss) [reply tests/README.md]: yup

File: tests/README.md (hunk 5 above)
Comment (MrKevinWeiss) [reply tests/README.md]: yup

File: tests/README.md (hunk 6 above)
Comment (MrKevinWeiss) [reply tests/README.md]: yup, those darn native english speakers.

File: README.md (hunk 7 above)
Comment (MrKevinWeiss) [reply README.md]: Testing and Changing RIOT Versions

File: tests/periph_i2c/README.md (hunk 8 above)
Comment (MrKevinWeiss) [reply tests/periph_i2c/README.md]: yup

File: tests/periph_i2c/README.md (hunk 2 above)
Comment (MrKevinWeiss) [reply tests/periph_i2c/README.md]: yup

File: tests/periph_i2c/README.md (hunk 1 above)
Comment (MrKevinWeiss) [reply tests/periph_i2c/README.md]: Not really, they don't need to if you are using non-isolated USB.  It would be safer to say that...

**Comment by MrKevinWeiss** [comment]
@smlng Can I squash?

File: README.md [hunk 9]
```diff
@@ -41,6 +41,10 @@ Besides the applications that are written in C, it is recommend to use Python
 to write tools or interfaces related to running tests. That allows for simple
 integration with the RF test scripts (and re-usability in general).
 
+## Testing and Changing RIOT Versions
+
+To test a pull request on RIOT go into the RIOT subdirectory and change to the desired state.  Alternatively, the `RIOTBASE=<path_to_riot_directory>` can be used.
```
Comment (smlng) [inline README.md]: I wouldn't say Pull Request, bc PRs are not part of a standard clone of RIOT (nor are say of the submodule) so you cannot simply change to a PR w/o changing the git-config of the submodule to also fetch PRs (which, again, is not done by default).

Comment (MrKevinWeiss) [reply README.md]: OK, version then?

Comment (MrKevinWeiss) [reply README.md]: Also I will fix the commit message

**Comment by smlng** [comment]
please squash
